### ISSUE TITLE
Linker warning

### ISSUE THREAD

**Issue by duncanspumpkin** [OpenLoco/OpenLoco#2187]
**Version information:**
 - Platform: Linux
 - Version: 23.10
 - Build/commit: [e.g. f08479f]

**Describe the bug**
```
/usr/bin/ld: warning: openloco_text_section.o: missing .note.GNU-stack section implies executable stack
/usr/bin/ld: NOTE: This behaviour is deprecated and will be removed in a future version of the linker
```
There is a linker warning with newer ld's.

Easily fixed by adding the following to our objcopy command:
```
 --add-section \".note.GNU-stack\"=/dev/null --set-section-flags \".note.GNU-stack\"=contents,readonly
```

**Comment by janisozaur** [comment]
We already link using our custom linker script, we can add this section there

**Comment by janisozaur** [comment]
https://github.com/SELinuxProject/selinux-kernel/blob/196e95aa8305aecafc4e1857b7d3eff200d953b6/arch/mips/vdso/vdso.lds.S#L63

I will try addressing this today

**Comment by janisozaur** [comment]
We already have that: https://github.com/OpenLoco/OpenLoco/blob/935b4c6dd38e0df6ac3186e93a80eebf7f22ae28/distribution/linux/ld_script_i386.xc#L217

I'll have to check what else do we need

**Comment by duncanspumpkin** [comment]
> We already have that:
> 
> https://github.com/OpenLoco/OpenLoco/blob/935b4c6dd38e0df6ac3186e93a80eebf7f22ae28/distribution/linux/ld_script_i386.xc#L217
> 
> I'll have to check what else do we need

I think it might be about the input into the linker rather than what we are doing in the linker. That's why adding it to the objcopy works

**Comment by janisozaur** [comment]
Yeah, I know, but something has to generate this section already in other executable binaries. We use only a minor modification to the linker script, so I would expect the section to be present __waves hand__ in there. I need to dig a bit deeper.

The warning itself is ambiguous as well - what exactly is not supported and will be removed? Will this section be forced into binaries? Will the binary no longer work? Due to the missing section or due to the stack being executable?
